### DISCUSSION TITLE
relatableQuery working but relatableUsers etc does nothing

### DISCUSSION THREAD

**Discussion by RealityGamin** [laravel/nova-issues#4381] [Q&A] [answered]
- Laravel Version: master
- Nova Version: 4.7.0
- PHP Version: 8.1

### Description:
It seems like relatableUsers is not working, but relatableQuery works fine. 

When using the example below, nothing happens and the resource loads fine, no network or console errors

### Detailed steps to reproduce the issue on a fresh Nova installation:

**Course - Resource**
```
class Course extends Resource
{
    /**
     * The model the resource corresponds to.
     *
     * @var string
     */
    public static $model = \App\V2\VCourse::class;
    
    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @return array
     */
    public function fields(NovaRequest $request)
    {
        return [
            ID::make()->sortable(),
            BelongsTo::make('User', 'user', 'App\Nova\User')->nullable()->sortable(),
        ];
    }
   /**
     * Build a "relatable" query for the given resource.
     *
     * This query determines which instances of the model may be attached to other resources.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public static function relatableUsers(NovaRequest $request, $query){
        dd('i cant see this message!');
    }
}
```

**VCourse - Model**
```
class VCourse extends Model
{
	use SoftDeletes;
    
    public function user(){
        return $this->belongsTo(VUser::class, 'vuser_id', 'id');
    }
}
```

When using **relatableQuery - on User Resource** dd($request()->all()) im getting:
```
^ array:5 [
  "editing" => "true"
  "editMode" => "update"
  "viaResource" => null
  "viaResourceId" => null
  "viaRelationship" => null
]
```

When loading the edit course page, but I feel like it should be showing the relationship related right? or am I missing something here?

**Comment by crynobone** [comment, marked as answer]
The dynamic filtering method uses the plural of the base model name, which in this case makes it `relatableVUsers`

**Reply by RealityGamin** [reply]
You are absolutely wonderful, it worked.

In the documentation, I understood that It needs to match the laravel resource name not the model name, I appreciate it thank you.
